pyccl/base/deprecations.py

```diff
@@ -15,45 +15,51 @@ def deprecated(new_function=None):
     """
     def decorator(func):
         @functools.wraps(func)
-        def new_func(*args, **kwargs):
-            s = "The function {} is deprecated.".format(func.__name__)
+        def wrapper(*args, **kwargs):
+            s = f"The function {func.__qualname__} is deprecated."
             if new_function:
-                s += " Use {} instead.".format(new_function.__name__)
+                s += f" Use {new_function.__qualname__} instead."
             warnings.warn(s, CCLDeprecationWarning)
             return func(*args, **kwargs)
-        return new_func
+        return wrapper
```

Comment:
Clear case of mission drift.

Reply:
haha no I actually brought that in from one of my other branches. I pasted it into here, I didn't write it.